### PR TITLE
nimble/iso: Add ISO Rx datapath and extend ISO Broadcast API

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This adds ISO Rx datapath and along with Broadcast Sink ISO API
and implementation.
This also extends btshell application with Broadcast ISO related commands.

Depends on: https://github.com/apache/mynewt-nimble/pull/1700

### CODE REVIEW

File: nimble/host/src/ble_iso.c [hunk 1]
```diff
@@ -354,4 +490,449 @@
...
+    return 0;
+}
+#endif /* BLE_ISO_BROADCAST_SINK */
+
+int
+ble_iso_data_path_setup(const struct ble_iso_data_path_setup_params *param)
+{
+    struct ble_hci_le_setup_iso_data_path_rp rp;
+    struct ble_hci_le_setup_iso_data_path_cp *cp;
+    uint8_t buf[sizeof(*cp) + UINT8_MAX]; /* FIXME */
```
[the remark this answers is not in the document]
Comment (MariuszSkamra) [reply nimble/host/src/ble_iso.c]: done

File: nimble/include/nimble/hci_common.h [hunk 2]
```diff
@@ -2087,39 +2087,53 @@ struct hci_data_hdr
...
 #define BLE_HCI_ISO_LENGTH(l)           ((l) & BLE_HCI_ISO_LENGTH_MASK)
+#define BLE_HCI_ISO_SDU_LENGTH(l)       ((l) & BLE_HCI_ISO_SDU_LENGTH_MASK)
+#define BLE_HCI_ISO_PKT_STATUS_FLAG(l)  (((l) & BLE_HCI_ISO_PKT_STATUS_FLAG_MASK) >> 14)
 
 #define BLE_HCI_ISO_PB_FIRST            (0)
 #define BLE_HCI_ISO_PB_CONTINUATION     (1)
 #define BLE_HCI_ISO_PB_COMPLETE         (2)
 #define BLE_HCI_ISO_PB_LAST             (3)
 
+#define BLE_HCI_ISO_PKT_STATUS_VALID    0b00
```
Comment (rymanluk) [inline nimble/include/nimble/hci_common.h:2111]: lets just follow the style in the file here `0x00`

File: nimble/host/include/host/ble_iso.h [hunk 3]
```diff
@@ -45,6 +53,36 @@ struct ble_iso_big_desc
     uint16_t conn_handle[MYNEWT_VAL(BLE_MAX_BIS)];
 };
 
+/** @brief Received ISO Data status possible values */
+enum ble_iso_rx_data_status {
+    /** The complete SDU was received correctly. */
+    BLE_ISO_DATA_STATUS_VALID = 0,
```
Comment (rymanluk) [inline nimble/host/include/host/ble_iso.h:59]: maybe BLE_ISO_DATA_STATUS_VALID =  BLE_HCI_ISO_PKT_STATUS_VALID  ?

File: nimble/include/nimble/hci_common.h (hunk 2 above)
Comment (MariuszSkamra) [reply nimble/include/nimble/hci_common.h]: done

**Comment by MariuszSkamra** [comment]
* Fixed coding style issues
* The remaining coding style issues discovered by CI are false positives

File: nimble/host/src/ble_hs_hci_evt.c [hunk 4]
```diff
@@ -779,6 +789,39 @@ ble_hs_hci_evt_le_terminate_big_complete(uint8_t subevent, const void *data,
...
 #endif
 
+#if MYNEWT_VAL(BLE_ISO_BROADCAST_SINK)
+static int
+ble_hs_hci_evt_le_big_sync_established(uint8_t subevent, const void *data,
+                                       unsigned int len)
+{
+    const struct ble_hci_ev_le_subev_big_sync_established *ev = data;
+
+    if (len < sizeof(*ev) ||
```
Comment (rymanluk) [inline nimble/host/src/ble_hs_hci_evt.c:799]: `len < sizeof(*ev) |` seems to be reduntant

Comment (rymanluk) [reply nimble/host/src/ble_hs_hci_evt.c:799]: it is ok actually to make sure next check reads from proper area

**Pull request merged**